docs/user_guide/mitigation.rst

```diff
@@ -68,6 +68,259 @@ Postprocessing
 
 .. currentmodule:: fairlearn.postprocessing
 
+Fairlearn currently supports one postprocessing technique that is based on the
+paper *Equality of Opportunity in Supervised Learning* [#3]_. Unlike other
+mitigation techniques :class:`ThresholdOptimizer` is built to satisfy the
+specified fairness criteria exactly and with no remaining disparity.
+In many cases this comes at the expense of performance, for example, with
+significantly lower accuracy. Regardless, it is a useful data point to compare
+results with.
+
+:class:`ThresholdOptimizer` expects an estimator that provides it with scores.
+While the output of :class:`ThresholdOptimizer` is binary, the input need not
+be. In fact, real valued input, e.g. from a regressor, provides it with many
+more options to create thresholds. For :math:`n` input data points with
+:math:`m \leq n` different score values it has :math:`m+1` different
+thresholds. At each threshold one can create one of two thresholding rules,
+i.e. functions that indicate which data points get label :code:`1` based on
+their score.
+
+We can think of each thresholding rule as a binary classifier and determine
+its true positive rate and false positive rate.
+The code below visualizes the threshold selection process with an ROC curve.
+The ROC curves consist of the true and false positive rates for each
+of the thresholding rules, with separate ones per sensitive feature value.
+Note that the plot omits points that are within the convex hull of points.
+
+.. plot::
+    :include-source:
+
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from fairlearn.postprocessing import ThresholdOptimizer, plot_threshold_optimizer
+    >>> from sklearn.datasets import fetch_openml
+    >>> import pandas as pd
+    >>> data = fetch_openml(data_id=1590, as_frame=True)
+    >>> X = pd.get_dummies(data.data)
+    >>> y = (data.target == '>50K') * 1
+    >>> sex = data.data['sex']
+    >>> logistic_regression = LogisticRegression()
+    >>> threshold_optimizer = ThresholdOptimizer(
+    ...     estimator=logistic_regression,
+    ...     constraints="equalized_odds",
+    ...     objective="accuracy_score")
+    >>> threshold_optimizer.fit(X, y, sensitive_features=sex)
+    ThresholdOptimizer(constraints='equalized_odds', estimator=LogisticRegression())
+    >>> threshold_optimizer.interpolated_thresholder_._print_thresholds()
```

Comment:
an example shouldn't be using private API. We either need to make it public, or instead write a little function in this example which does the printing for us (my preferred way)

Reply:
I wrote that function specifically for this user guide actually. Even if I wrote it just in this snippet it would still have to access private members. If people are fine with it I can make `_print_thresholds()` public by removing the underscore. Wdyt?

Reply:
This doesn't use the private API anymore since I used your suggestion from another comment. Thanks!